x-pack/plugins/lens/public/xy_visualization/to_expression.ts

```diff
@@ -214,10 +215,9 @@ export const buildExpression = (
                       : [],
                     position: !state.legend.isInside ? [state.legend.position] : [],
                     isInside: state.legend.isInside ? [state.legend.isInside] : [],
-                    legendSize:
-                      !state.legend.isInside && state.legend.legendSize
-                        ? [state.legend.legendSize]
-                        : [],
+                    legendSize: !state.legend.isInside
+                      ? [state.legend.legendSize ?? DEFAULT_LEGEND_SIZE]
+                      : [],
```

Comment:
This logic was added in https://github.com/elastic/kibana/pull/128255.

It ignores the legend size setting if the legend is not set to "inside" (uses "auto" instead).

I'm not sure this is compatible with the intent of _this_ PR which is to stop using legend size "auto" in visualizations.

Thoughts?

@Kunzetsov @flash1293 @ghudgins

Reply:
I think it should be fine - AFAIK this change was made because the legend size doesn't have an effect if "inside" is used. The reasoning behind not having an "auto" legend size is that it makes it hard to have a uniform dashboard with legends across panels lining up. For the inline legends this is not a concern.

Reply:
@andrewctate, I've removed all the changes to show you the problem I was solving. Before my PR, the legend size could be specified, but was not applied to the legend content, if the position was `inside`.
So, I've removed the control for legend size definition, if the legend position is equal to `inside`. If you are able to make the legend size applyable to the legend, you can remove all the code, I've added. 

## outside
<img width="741" alt="Screenshot 2022-05-04 at 10 33 58" src="https://user-images.githubusercontent.com/22456368/166639351-905120ec-c15f-42b6-be74-8d6b8e95a461.png">

## inside
<img width="741" alt="Screenshot 2022-05-04 at 10 34 09" src="https://user-images.githubusercontent.com/22456368/166639373-548acd12-754c-4a02-9a31-644cedbb7457.png">


cc @flash1293